src/sentry/db/models/utils.py

```diff
@@ -77,8 +77,11 @@ def slugify_instance(
 
     setattr(inst, field_name, base_value)
 
-    # We don't need to further mutate if we're unique at this point
-    if not base_qs.filter(**{f"{field_name}__iexact": base_value}).exists():
+    # We don't need to further mutate if the value is unique or not entirely numeric
+    if (
+        not base_qs.filter(**{f"{field_name}__iexact": base_value}).exists()
+        and not base_value.isdigit()
+    ):
```

Comment:
Update `slugify_instance` which is used to automatically set `slug` using `name` if creating a new resource where `slug` is not set. By not returning if the slug is entirely numeric, we go to the logic below which adds a suffix to the slug